qiita_db/sql_connection.py

```diff
@@ -460,176 +397,199 @@ def execute_fetchall(self, sql, sql_args=None):
 
         return result
 
-    def _check_queue_exists(self, queue_name):
-        """Checks if queue `queue_name` exists in the handler
 
-        Parameters
-        ----------
-        queue_name : str
-            The name of the queue
+class Transaction(object):
+    """Encapsulates a DB transaction
 
-        Returns
-        -------
-        bool
-            True if queue `queue_name` exist in the handler. False otherwise.
-        """
-        return queue_name in self.queues
+    A transaction is defined by a series of consecutive queries that need to
+    be applied to the database as a single block.
+
+    Parameters
+    ----------
+    name : str
+        Name of the transaction.
+    """
 
-    def create_queue(self, queue_name):
-        """Add a new queue to the connection
+    _regex = re.compile("^{(\d+):(\d+):(\d+)}$")
 
-        Parameters
-        ----------
-        queue_name : str
-            Name of the new queue
+    def __init__(self, name):
+        # The name is useful for debugging, since we can identify the
+        # failed queue in errors
+        self._name = name
+        self._queries = []
+        self._results = []
```

Comment:
What does memory use look like for this list on large transaction blocks? Since this is a dense representation of all SQL queries' results, this is a concern.